docs/ecs_guide.md

```diff
@@ -0,0 +1,269 @@
+
+If you specify a task definition [ARN (Amazon Resource Name)](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference-arns.html) in the work pool, the worker will use that ARN when spinning up the EC2 instance, rather than creating a new ECS Fargate container.
+
+
+You can use either EC2 or Fargate as the capacity provider. Fargate simplifies initiation and provides containerization, but lengthens infrastructure setup time for each flow run. Using EC2 for the ECS cluster can reduce setup time. In this example, we will show how to ue Fargate.
+<hr>
+
+!!! tip
+    If you prefer IaC (Infrastructure as Code) check out this [terraform module](https://github.com/PrefectHQ/prefect-recipes/tree/main/devops/infrastructure-as-code/aws/tf-prefect2-ecs-worker) to provision an ECS cluster with a worker.
+
+## Prerequisites
+Before you begin, make sure you have:
+
+- An AWS account with permissions to create ECS services and IAM roles.
+- The AWS CLI installed on your local machine. You can [download it from the AWS website](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html).
+- An [ECS Cluster](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/clusters.html) to host both the worker and the flow runs it submits.
+- A [VPC](https://docs.aws.amazon.com/vpc/latest/userguide/what-is-amazon-vpc.html) configured for your ECS tasks. A VPC is a good idea if using EC2 and required if using Fargate. TK is this true?
+
+## Step 1: Set Up an ECS work pool
+Before setting up the worker, create a simple [work pool](https://docs.prefect.io/latest/concepts/work-pools/#work-pool-configuration) of type ECS for the worker to pull work from.
+
+Create a work pool from the Prefect UI or CLI:
+
+`prefect work-pool create --type ecs my-ecs-pool`
+
+Configure the VPC and ECS Cluster for your Work Pool via the UI:
+![VPC](img/VPC_UI.png)
+
+Configuring custom fields is easiest from the UI.
+
+![ECSCluster](img/ECSCluster_UI.png)
+
+!!! Warning
+    You need to have a VPC specified for your work pool if you are using AWS Fargate.
+    
+![Launch](img/LaunchType_UI.png)
+<sub>Pictures of UI Cropped
+
+Next, set up an Prefect ECS worker that will discover and pull work from this work pool.
+## Step 2: Set up an ECS worker
+Start a Prefect worker in your ECS cluster.
+
+Create a Docker image for your Prefect worker, which you can build and push to the Amazon ECR registry. TK do you have to?
+
+For example, a minimal Dockerfile for an ECS worker could look like:
+```Dockerfile
+FROM prefecthq/prefect:2-python3.10
+RUN pip install s3fs prefect-aws
+```
+
+### Step 3: Create an IAM Role for the ECS task
+
+To create an IAM role for the ECS task using the AWS CLI, follow these steps: 
+
+1. **Create a Trust Policy**
+
+    The Trust Policy will specify that ECS can assume the role. 
+    
+    Save this policy to a file, such as `ecs-trust-policy.json`:
+
+    ```json
```

Comment:
We should also definitely link out to AWS documentation for doing this from the console

Reply:
Added a link